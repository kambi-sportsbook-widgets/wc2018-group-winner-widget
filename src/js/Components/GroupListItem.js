import React from 'react'
import PropTypes from 'prop-types'
import { OutcomeButton } from 'kambi-widget-components'
import styles from './GroupListItem.scss'



const GroupListItem = ({ participant, outcomes, flagUrl, handleClick }) => {   
   /**
    * Removes images with broken urls
    */   
   const handleBrokenUrl = () => {
      this.img.style.display = 'none'
   }
   
   return (
     <div>
        <li className={styles.row}>
           {
             flagUrl ? (
               <div className={styles.flag} onClick={handleClick} ref={(img) => { this.img = img }}>
                 <img
                   role="presentation"
                   src={flagUrl}
                   onError={() => this.handleBrokenUrl()}
                 />
               </div>
             ) : null
           }
           <span className={styles.participant} onClick={handleClick}>
             {participant}
           </span>
           {
               outcomes.map(outcome => (
                  <div className={styles.button}>
                     <OutcomeButton outcome={outcome} label={false} outlineStyle={true} />
                  </div>
               ))
           }
           
        </li>
     </div>
   )
}



/**
 * participant { node } list item
 * flagUrl { string } url to item flag
 * outcomes { array } outcomes for betOffer
 * handleClick { func } callback for item click
 */
GroupListItem.propTypes = {
  participant: PropTypes.node.isRequired,
  flagUrl: PropTypes.string,
  outcomes: PropTypes.arrayOf(PropTypes.shape()).isRequired,
  handleClick: PropTypes.func.isRequired,
}

export default GroupListItem