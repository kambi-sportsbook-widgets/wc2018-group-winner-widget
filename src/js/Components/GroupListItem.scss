.row {
   display: flex;
   align-items: center;
   height: 40px;
   cursor: pointer;
   list-style: none;
   justify-content: space-between;
   color: #fff;
   .participantWrapper {
      flex-grow: 1;
      flex-shrink: 0;
      flex-basis: auto;
      z-index: 1;
      display: flex;
      flex-wrap: wrap;
   }
   .flag {
      line-height: 1;
      padding-right: 12px;
      width: 24px;
      flex-grow: 0;
      flex-shrink: 0;
      flex-basis: 36px;
      img {
         width: 24px;
         height: 24px;
      }
   }
   .participant {
      font-size: 14px;
      overflow: hidden;
      text-overflow: ellipsis;
      white-space: nowrap;
      display: block;
      max-width: 140px;
   }
   .button,
   .emptyButton {
      flex-grow: 0;
      flex-shrink: 0;
      flex-basis: 60px;
      display: flex;
      height: 36px;
      margin-left: .5rem;
      z-index: 1;
   }
}
