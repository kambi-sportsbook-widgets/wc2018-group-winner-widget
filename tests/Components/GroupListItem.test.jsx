/* eslint-env jest */
import React from 'react';
import ReactShallowRenderer from 'react-test-renderer/shallow'
import { shallow } from 'enzyme'

import GroupListItem from '../../src/js/Components/GroupListItem';


const mockData = {
   participant: 'Uruguay',
   outcomes: [{
      betOfferId: 2104811235,
      changedDate:"2018-02-07T11:55:05Z",
      englishLabel:"Uruguay",
      id:2373848420,
      label:"Uruguay",
      odds:2100,
      oddsAmerican:"110",
      oddsFractional:"11/10",
      participant:"Uruguay",
      participantId:1000000155,
      status:"OPEN",
      type:"OT_UNTYPED"
   }, {
      betOfferId: 2104811236,
      changedDate:"2018-02-07T11:55:05Z",
      englishLabel:"Uruguay",
      id:2373848422,
      label:"Uruguay",
      odds:1180,
      oddsAmerican:"-590",
      oddsFractional:"1/6",
      participant:"Uruguay",
      participantId:1000000155,
      status:"OPEN",
      type:"OT_UNTYPED"
   }],
   flagUrl: "assets/icons/uruguay.svg",
   handleClick: jest.fn()
}

const mockOnlyOneOutcome = {
   participant: 'Uruguay',
   outcomes: [{
      betOfferId: 2104811236,
      changedDate:"2018-02-07T11:55:05Z",
      englishLabel:"Uruguay",
      id:2373848422,
      label:"Uruguay",
      odds:1180,
      oddsAmerican:"-590",
      oddsFractional:"1/6",
      participant:"Uruguay",
      participantId:1000000155,
      status:"OPEN",
      type:"OT_UNTYPED"
   }],
   flagUrl: "assets/icons/uruguay.svg",
   handleClick: jest.fn()
}

let renderer;

describe('GroupListItem DOM Rendering', () => {

   const { participant, outcomes, flagUrl, handleClick } = mockData

   beforeEach(() => {
      renderer = new ReactShallowRenderer();
   });

   it('renders correctly', () => {
      expect(renderer.render(
         <GroupListItem
            participant={participant}
            outcomes={outcomes}
            flagUrl={flagUrl}
            handleClick={handleClick}
         />
      )).toMatchSnapshot();
   });

   it('renders correctly withouth flagUrl', () => {
      expect(renderer.render(
         <GroupListItem
            participant={participant}
            outcomes={outcomes}
            handleClick={handleClick}
         />
      )).toMatchSnapshot();
   });

   it('handles a broken imgLink correctly', () => {
      expect(renderer.render(
         <GroupListItem
            participant={participant}
            outcomes={outcomes}
            flagUrl={'assets/icons/incorrecturl.svg'}
            handleClick={handleClick}
         />
      )).toMatchSnapshot();
   });

   it('handles only receiving one outcome per participant', () => {
      expect(renderer.render(
         <GroupListItem
            participant={participant}
            outcomes={outcomes}
            flagUrl={'assets/icons/incorrecturl.svg'}
            handleClick={handleClick}
         />
      )).toMatchSnapshot();
   });

});
