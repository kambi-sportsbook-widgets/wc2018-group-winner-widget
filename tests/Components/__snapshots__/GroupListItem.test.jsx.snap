// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GroupListItem DOM Rendering handles a broken imgLink correctly 1`] = `
<li
  className="row"
>
  <div
    className="participantWrapper"
  >
    <div
      className="flag"
      onClick={[MockFunction]}
    >
      <img
        onError={[Function]}
        role="presentation"
        src="assets/icons/incorrecturl.svg"
      />
    </div>
    <span
      className="participant"
      onClick={[MockFunction]}
    >
      Uruguay
    </span>
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811235,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848420,
          "label": "Uruguay",
          "odds": 2100,
          "oddsAmerican": "110",
          "oddsFractional": "11/10",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811236,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848422,
          "label": "Uruguay",
          "odds": 1180,
          "oddsAmerican": "-590",
          "oddsFractional": "1/6",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
</li>
`;

exports[`GroupListItem DOM Rendering handles only receiving one outcome per participant 1`] = `
<li
  className="row"
>
  <div
    className="participantWrapper"
  >
    <div
      className="flag"
      onClick={[MockFunction]}
    >
      <img
        onError={[Function]}
        role="presentation"
        src="assets/icons/incorrecturl.svg"
      />
    </div>
    <span
      className="participant"
      onClick={[MockFunction]}
    >
      Uruguay
    </span>
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811235,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848420,
          "label": "Uruguay",
          "odds": 2100,
          "oddsAmerican": "110",
          "oddsFractional": "11/10",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811236,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848422,
          "label": "Uruguay",
          "odds": 1180,
          "oddsAmerican": "-590",
          "oddsFractional": "1/6",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
</li>
`;

exports[`GroupListItem DOM Rendering renders correctly 1`] = `
<li
  className="row"
>
  <div
    className="participantWrapper"
  >
    <div
      className="flag"
      onClick={[MockFunction]}
    >
      <img
        onError={[Function]}
        role="presentation"
        src="assets/icons/uruguay.svg"
      />
    </div>
    <span
      className="participant"
      onClick={[MockFunction]}
    >
      Uruguay
    </span>
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811235,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848420,
          "label": "Uruguay",
          "odds": 2100,
          "oddsAmerican": "110",
          "oddsFractional": "11/10",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811236,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848422,
          "label": "Uruguay",
          "odds": 1180,
          "oddsAmerican": "-590",
          "oddsFractional": "1/6",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
</li>
`;

exports[`GroupListItem DOM Rendering renders correctly withouth flagUrl 1`] = `
<li
  className="row"
>
  <div
    className="participantWrapper"
  >
    <span
      className="participant"
      onClick={[MockFunction]}
    >
      Uruguay
    </span>
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811235,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848420,
          "label": "Uruguay",
          "odds": 2100,
          "oddsAmerican": "110",
          "oddsFractional": "11/10",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
  <div
    className="button"
  >
    <OutcomeButton
      betslipDefaultStake={null}
      betslipTrackingName={null}
      betslipUpdateMode={null}
      event={undefined}
      label={false}
      onSelection={null}
      outcome={
        Object {
          "betOfferId": 2104811236,
          "changedDate": "2018-02-07T11:55:05Z",
          "englishLabel": "Uruguay",
          "id": 2373848422,
          "label": "Uruguay",
          "odds": 1180,
          "oddsAmerican": "-590",
          "oddsFractional": "1/6",
          "participant": "Uruguay",
          "participantId": 1000000155,
          "status": "OPEN",
          "type": "OT_UNTYPED",
        }
      }
      outlineStyle={true}
      updateOdds={false}
    />
  </div>
</li>
`;
